app/assets/javascripts/discourse/app/models/site.js

```diff
@@ -139,6 +139,21 @@ const Site = RestModel.extend({
       return newCategory;
     }
   },
+
+  allowPersonalMessages(user) {
+    if (!user) {
+      return false;
+    }
+
+    return (
+      user.staff ||
+      user.isInAnyGroups(
+        this.siteSettings.personal_message_enabled_groups
```

Comment:
can this be empty? If yes we should `filter(Boolean)`

```javascript
"".split("|") // [""]
````

Reply:
No it should never be empty, I will need to make sure the site setting has validation for that

Reply:
Ok this is done